src/rpc/mining.cpp

```diff
@@ -801,7 +802,11 @@ static RPCHelpMan getblocktemplate()
         {
             MillisecondsDouble checktxtime{std::chrono::minutes(1)};
             while (tip == hashWatchedChain && IsRPCRunning()) {
-                tip = miner.waitTipChanged(hashWatchedChain, checktxtime).hash;
+                std::optional<BlockRef> maybe_tip{miner.waitTipChanged(hashWatchedChain, checktxtime)};
+                if (!maybe_tip) {
+                    throw JSONRPCError(RPC_CLIENT_NOT_CONNECTED, "Shutting down");
```

Comment:
Maybe elaborate the message that we could be here if no tip was connected within the timeout: "No tip within timeout or shutting down".

Reply:
I'm introducing a new `RPC_SHUTDOWN_ERROR` which makes sense here. I don't think the message itself matters, since the user knows they shut down the node - and the exact details of where and why the RPC call fails aren't important.